page/sw.js

```diff
@@ -25,59 +25,22 @@ self.addEventListener('push', function(event)  {
         console.log("Service worker just got:", content);
         // Send the event to the parent pages.
         event.waitUntil(
-          self.clients.matchAll()
-           .then(clientList => {
-              let sent = false;
-              console.debug("Service worker found clients",
-                    JSON.stringify(clients));
-             clientList.forEach(client => {
-                 console.debug("Service worker sending to client...", client);
-                 sent = true;
-                 client.postMessage({'type':'content','content':content});
-             });
-             if (sent == false) {
-                 throw new Error("No valid client to send to.");
-             }
-           })
-           .catch(err => {
-              console.error("Service worker couldn't send message: ", err);
-           })
+          self.registration.showNotification("Got message", {
+            body: content,
+            icon: "icon.png",
+            //tag: "collapse tag",  // This will collapse all notifications with this tag name
+          })
```

Comment:
I may add the message function back as a comment. You're absolutely right that for this demo we can just show the notification and be done with it, but Push is more than just that, and I'd love folks realizing that there's so much more they can do.